rts/Lua/LuaOpenGL.cpp

```diff
@@ -4657,15 +4664,299 @@ int LuaOpenGL::GetMapRendering(lua_State* L)
 			lua_pushboolean(L, mapRendering->voidGround);
 			return 1;
 		} break;
+		case hashString("specularLighting"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			lua_pushboolean(L, smfMap->GetSpecularTexture() != 0);
+			return 1;
+		} break;
+		case hashString("splatDetailTexture"): {
```

Comment:
Should be named correctly along with pretty much everything here.
This doesn't return a texture but a Boolean representing if the texture exists.